pkg/datapath/linux/node_linux_test.go

```diff
@@ -422,281 +398,6 @@ func (s *linuxPrivilegedBaseTestSuite) TestAuxiliaryPrefixes(t *testing.T) {
-	s.commonNodeUpdateEncapsulation(t, false, func(*nodeTypes.Node) bool { return true })
-}
-
-func (s *linuxPrivilegedBaseTestSuite) commonNodeUpdateEncapsulation(t *testing.T, encap bool, override func(*nodeTypes.Node) bool) {
-	ip4Alloc1 := cidr.MustParseCIDR("5.5.5.0/24")
-	ip4Alloc2 := cidr.MustParseCIDR("6.6.6.0/24")
-	ip6Alloc1 := cidr.MustParseCIDR("2001:aaaa::/96")
-	ip6Alloc2 := cidr.MustParseCIDR("2001:bbbb::/96")
-
-	externalNodeIP1 := net.ParseIP("4.4.4.4")
-	externalNodeIP2 := net.ParseIP("8.8.8.8")
-
-	dpConfig := DatapathConfiguration{HostDevice: dummyHostDeviceName}
-	log := hivetest.Logger(t)
-	linuxNodeHandler := newNodeHandler(log, dpConfig, nodemapfake.NewFakeNodeMapV2(), new(mockEnqueuer))
-
-	require.NotNil(t, linuxNodeHandler)
-	linuxNodeHandler.OverrideEnableEncapsulation(override)
-	nodeConfig := s.nodeConfigTemplate
-	nodeConfig.EnableEncapsulation = encap
-	err := linuxNodeHandler.NodeConfigurationChanged(nodeConfig)
-	require.NoError(t, err)
-
-	// nodev1: ip4Alloc1, ip6alloc1 => externalNodeIP1
-	nodev1 := nodeTypes.Node{
-		Name:      "node1",
-		ClusterID: 11,
-		IPAddresses: []nodeTypes.Address{
-			{IP: externalNodeIP1, Type: nodeaddressing.NodeInternalIP},
-		},
-	}
-
-	if s.enableIPv4 {
-		nodev1.IPv4AllocCIDR = ip4Alloc1
-	}
-	if s.enableIPv6 {
-		nodev1.IPv6AllocCIDR = ip6Alloc1
-	}
-
-	err = linuxNodeHandler.NodeAdd(nodev1)
-	require.NoError(t, err)
-
-	if s.enableIPv4 {
-		underlayIP, err := tunnel.TunnelMap().GetTunnelEndpoint(cmtypes.MustAddrClusterFromIP(ip4Alloc1.IP))
-		require.NoError(t, err)
-		require.True(t, underlayIP.Equal(externalNodeIP1))
-
-		foundRoute, err := linuxNodeHandler.lookupNodeRoute(ip4Alloc1, false)
-		require.NoError(t, err)
-		require.NotNil(t, foundRoute)
-	}
-
-	if s.enableIPv6 {
-		underlayIP, err := tunnel.TunnelMap().GetTunnelEndpoint(cmtypes.MustAddrClusterFromIP(ip6Alloc1.IP))
-		require.NoError(t, err)
-		require.True(t, underlayIP.Equal(externalNodeIP1))
-
-		foundRoute, err := linuxNodeHandler.lookupNodeRoute(ip6Alloc1, false)
-		require.NoError(t, err)
-		require.NotNil(t, foundRoute)
-	}
-
-	// nodev2: ip4Alloc1, ip6alloc1 => externalNodeIP2
-	nodev2 := nodeTypes.Node{
-		Name:      "node1",
-		ClusterID: 11,
-		IPAddresses: []nodeTypes.Address{
-			{IP: externalNodeIP2, Type: nodeaddressing.NodeInternalIP},
-		},
-	}
-
-	if s.enableIPv4 {
-		nodev2.IPv4AllocCIDR = ip4Alloc1
-	}
-	if s.enableIPv6 {
-		nodev2.IPv6AllocCIDR = ip6Alloc1
-	}
-
-	err = linuxNodeHandler.NodeUpdate(nodev1, nodev2)
-	require.NoError(t, err)
-
-	// alloc range v1 should map to underlay2
-	if s.enableIPv4 {
-		underlayIP, err := tunnel.TunnelMap().GetTunnelEndpoint(cmtypes.MustAddrClusterFromIP(ip4Alloc1.IP))
-		require.NoError(t, err)
-		require.True(t, underlayIP.Equal(externalNodeIP2))
-
-		foundRoute, err := linuxNodeHandler.lookupNodeRoute(ip4Alloc1, false)
-		require.NoError(t, err)
-		require.NotNil(t, foundRoute)
-	}
```

Comment:
I understand that we can drop the testing for the tunnel map. But why not keep the testing for the routes?

Reply:
I was under the impression that `TestUpdateNodeRoute` was enough but you're right, `TestNodeUpdateEncapsulation` and `TestNodeUpdateEncapsulationWithOverride` exercise the exported methods `NodeUpdate` and `NodeDelete`m so they guarantee a better coverage. Fixed :+1:

Reply:
Could you please split the patch in two - one that removes *only* the tunnel map-related parts, and a second one that subsequently removes all the other redundant testing (with an explanation why they are redundant) ? 

This will make review much more obvious, and allows us to easily roll back if the tests turn out not to be redundant after all ;).

Reply:
Splitted the commit in multiple ones, to address each separate case differently. I've also realized there was a mistake in the current test suite that led to the execution of the same subtest twice (while another one was not called at all). Please review commit by commit for all the info :pray:

Reply:
thank you! :heart: